### DISCUSSION TITLE
JSoC- Proposal optical flow expansion

### DISCUSSION THREAD

**Discussion by gocari** [JuliaImages/Images.jl#1002] [JSoC]
Hello everyone, 

My name is Rilene and I would like to join the GSOC 2022 ecosystem community with JuliaImages.
I have been working with image processing for turbulence mitigation for a while and as of last six months I started converting some of our (research group) matlabs code to julia. 

Recently, I came across the work of Arijit Kar and his mentors (see https://summerofcode.withgoogle.com/archive/2018/projects/5996673603993600), which has introduced me to JSoC. The optical flow algorithm is very important to my field and therefore I would very much like to contribute to this project.

_ The project done by Mr. Kar comprised the development of:
1. Optical Flow Algorithms
    1.1. Lucas-Kanade method
    1.2. Farneback Dense Optical Flow method
2. Tracking Algorithms
    2.1. Boosting Tracking
    2.2. Multiple Instance Learning
    2.3. MedianFlow Tracking
    2.4. Tracking Learning Detection framework _

I would like to expand this project by means of adding two optical flow algorithm: 
1. Optical Flow Algorithms
    1.3.  Horn-Schunck Method (https://fzheng.me/2015/03/25/optical-flow/)
    1.4.  Sun & Black Method (https://files.is.tue.mpg.de/black/papers/SunECCV08old.pdf)

Regarding the second part, I am open to discussion. Here some ideas of what could be implemented:
- Further implementation of tracking algorithms
- Object detection algorithms?
- Registration/block matching algorithms?

(I would have availability to work on a 175h project.)

Looking forward to reading your comments and contributing to the community!

Rilene

**Comment by johnnychen94** [comment]
Unfortunately, I don't think we have the expertise here to mentor this tracking project, which is also why ImageTracking.jl is not under active maintenance. `@zygmuntszpak` might know this field but he's not available this summer.

If you want to contribute new algorithms to ImageTracking.jl, PRs are welcomed. But for the GSoC project, without the expertise we can't offer mentorship, you might need to propose a different project for GSoC.

> Registration/block matching algorithms?

This is indeed a missing functionality part of JuliaImages, I know @timholy has some registration codes in holylab https://github.com/HolyLab/RegisterCore.jl so I bet he'll be interested. I'm also interested in this but I don't have concrete algorithms in mind.

There's also https://github.com/peterkovesi/ImageProjectiveGeometry.jl developed by `@peterkovesi` a long time ago, it's long broken since he's retired, I would like to see this reimplemented using modern Julia and JuliaImages but never had a chance to do it myself.
